src/authenticate.ts

```diff
@@ -0,0 +1,23 @@
+import { DecodedToken } from './interfaces';
+import { CustomError } from './custom-error';
+import { verify } from 'jsonwebtoken';
+
+export function authenticate(token: string) {
+  const hasToken = !!token;
+  if (!hasToken) {
```

Comment:
```suggestion
  if (!token) {
```

Se o token for alguma coisa que avalia pra false no javascript (undefined, null, string vazia, ...), ele já entraria no if e jogaria o erro.